### DISCUSSION TITLE
Add YAML as an output format

### DISCUSSION THREAD

**Discussion by davetapley** [duckdb/duckdb#14337] [Ideas]
We have JSON on https://duckdb.org/docs/api/cli/output_formats.html, can we have YAML too?

Workaround is to `.mode json` and pipe to  `yq -P`, using https://github.com/duckdb/duckdb/discussions/14005#discussioncomment-10918638 to manage what is sent to `stdout`.

**Comment by themichaelusa** [comment]
This seems like a good candidate for a community extension. Read/Write YAML, and maybe throw TOML in there too!

**Reply by davetapley** [reply]
Great, I presume [start here](https://github.com/duckdb/extension-template/)?

Any tips? 😁

**Reply by themichaelusa** [reply]
Yeah that looks right! I'm planning to start on my first extension soon haha, so I don't have any more good advice than to look at the quack example in the extension-template

**Comment by dlaplante75** [comment]
Interested in this ! Keep us posted !

**Comment by davetapley** [comment]
FYI I won't be doing this, I realized I can `SELECT to_json(` then [`| yq -P`](https://stackoverflow.com/a/62798787/21115) as a workaround.

Happy to reopen if someone else wants to pick it up.

**Comment by pspeter3** [comment]
YAML and Markdown FrontMatter YAML would be great. You could use DuckDB to process CMS content.

**Comment by teaguesterling** [comment]
I decided I could use this too for a number of use cases, and went ahead to implement it. https://github.com/teaguesterling/duckdb_yaml (in full disclosure, technically Claude Code developed most of it). A PR to the  community extensions repo was recently merged, so it should be available soon.

**Reply by carlopi** [reply]
In duckdb v1.3.0:
```
INSTALL yaml FROM community;
LOAD yaml;
FROM '.github/workflows/LinuxRelease.yml';
```
just works! Nice!

(it do not work yet, even after `LOAD httpfs`, on remote files, I will open an issue)
